x-pack/plugins/security_solution/public/timelines/components/timeline/header/title_and_description.tsx

```diff
@@ -0,0 +1,219 @@
+          : i18n.SAVE_TIMELINE
+        : i18n.SAVE;
+
+    const calloutMessage = useMemo(() => i18n.UNSAVED_TIMELINE_WARNING(timelineType), [
+      timelineType,
+    ]);
+
+    const descriptionLabel = savedObjectId == null ? `${DESCRIPTION} (${OPTIONAL})` : DESCRIPTION;
+
+    return (
+      <>
+        {isSaving && (
+          <EuiProgress size="s" color="primary" position="absolute" data-test-subj="progress-bar" />
+        )}
+        <EuiModalHeader data-test-subj="modal-header">{modalHeader}</EuiModalHeader>
+
+        <Wrapper>
+          {showWarning && (
+            <EuiFlexItem grow={true}>
+              <EuiCallOut
+                title={calloutMessage}
+                color="danger"
+                iconType="alert"
+                data-test-subj="save-timeline-callout"
+              />
+              <EuiSpacer size="m" />
+            </EuiFlexItem>
+          )}
+          <EuiFlexItem grow={true}>
+            <EuiFormRow label={TIMELINE_TITLE}>
+              <Name
+                autoFocus={true}
+                className="timeline-modal-title"
+                disableTooltip={true}
+                disableAutoSave={true}
+                disabled={isSaving}
+                data-test-subj="save-timeline-name"
+                timelineId={timelineId}
+                timelineType={timelineType}
+                title={title}
+                updateTitle={updateTitle}
+                width="100%"
+                marginRight={10}
+              />
+            </EuiFormRow>
+            <EuiSpacer />
+          </EuiFlexItem>
+          <EuiFlexItem grow={true}>
+            <EuiFormRow label={descriptionLabel}>
+              <Description
+                data-test-subj="save-timeline-description"
+                description={description}
+                disableTooltip={true}
+                disableAutoSave={true}
+                disabled={isSaving}
+                timelineId={timelineId}
+                updateDescription={updateDescription}
+                isTextArea={true}
+                marginRight={0}
+              />
+            </EuiFormRow>
+            <EuiSpacer />
+          </EuiFlexItem>
+          <EuiFlexItem grow={false}>
+            <EuiFlexGroup justifyContent="flexEnd">
+              <EuiFlexItem grow={false} component="span">
+                {savedObjectId == null && showWarning ? (
+                  discardTimelineButton
+                ) : (
+                  <EuiButton
+                    fill={false}
+                    onClick={toggleSaveTimeline}
+                    isDisabled={isSaving}
+                    data-test-subj="close-button"
+                  >
+                    {i18n.CLOSE_MODAL}
+                  </EuiButton>
+                )}
+              </EuiFlexItem>
+              <EuiFlexItem grow={false} component="span">
+                <EuiButton
+                  isDisabled={title.trim().length === 0 || isSaving}
```

Comment:
Thanks for adding these checks! 😄 

Would you be willing to consider adding a condition that disables the button when the modal is launched in "edit" mode, but no changes have been made?

The rational for this change is clicking save when no mutations have been made (in edit mode) kicks off IO that updates the saved object, per the gif below:

![save-with-no-mutations](https://user-images.githubusercontent.com/4459398/97595885-c63dec80-19c9-11eb-9e80-21dedc74c4fd.gif)

Even though nothing _really_ changed, updating the saved object effects, for example, the Recent timelines widget on the Overview:

### Recent Timelines before save

<img width="343" alt="recent-timelines-before-save" src="https://user-images.githubusercontent.com/4459398/97596231-203eb200-19ca-11eb-8be0-56a30484c7f5.png">

### Recent Timelines after save

<img width="343" alt="recent-timelines-after-save" src="https://user-images.githubusercontent.com/4459398/97596258-2765c000-19ca-11eb-8900-f8150984754e.png">

Reply:
Yup, no problem. I'll open another PR for this.